src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -1485,6 +1529,170 @@ void hiopAlgFilterIPMQuasiNewton::outputIteration(int lsStatus, int lsNum, int u
+copy_vec_to_new_view(const ::std::string& name, const hiopVector* vec, sidre::Group* nlp_group)
+{
+  using IndType = sidre::IndexType;
+  const IndType size = vec->get_local_size();
+  sidre::View* dest = nlp_group->createViewAndAllocate(name, sidre::DOUBLE_ID, size);
+  
+  const auto stride(dest->getStride());
+  double *const dest_ptr(dest->getArray());
+  const double* arr = vec->local_data_host_const();
+  if(1==stride) {
+    ::std::copy(arr, arr+size, dest_ptr);
+  } else {
+    for(IndType i=0; i<size; ++i) {
+      dest_ptr[i*stride] = arr[i];
+    }
+  }  
+}
+void hiopAlgFilterIPMQuasiNewton::
+copy_vec_from_view(const ::std::string& name, hiopVector* vec, const sidre::Group* nlp_group)
+{
+  const sidre::View* view_const = nlp_group->getView(name);
+  if(!view_const) {
+    ::std::stringstream ss;
+    ss << "Could not find view '" << name << "in sidre::Group.\n";
+    throw ::std::runtime_error(ss.str());
+  }
+  // const_cast becase View does not have a const getArray()
+  sidre::View* view = const_cast<sidre::View*>(view_const);  
+  if(view) {
+    const hiop::size_type size = vec->get_local_size();
+    if(view->getNumElements() != size) {
+      ::std::stringstream ss;
+      ss << "Size mismatch for state/view '" << name << "' between hiop and sidre::View: " <<
+        " HiOp state is " << size << " doubles, while the view has " << view->getNumElements() <<
+        " double elements.\n";
+      throw ::std::runtime_error(ss.str());
+    }
+    double* arr_dest = vec->local_data_host();
+    const double *const arr_src = view->getArray();
+    const auto stride(view->getStride());
+    if(1==stride) {
+      ::std::copy(arr_src, arr_src+size, arr_dest);
+    } else {
+      for(hiop::index_type i=0; i<size; ++i) {
+        arr_dest[i] = arr_src[i*stride];
+      }
+    }    
+  } else {
+    assert(false && "const cast failed for Sidre::View");
+  }
+}
+
+void hiopAlgFilterIPMQuasiNewton::save_state_to_data_store(::axom::sidre::DataStore* ds)
+{
+  using IndType = sidre::IndexType;
+  sidre::Group* nlp_group = ds->getRoot()->createGroup("hiop solver");
+  
+  //create views for each member that needs to be saved
+  copy_vec_to_new_view("x", it_curr->get_x(), nlp_group);
+  copy_vec_to_new_view("d", it_curr->get_d(), nlp_group);
+  copy_vec_to_new_view("sxl", it_curr->get_sxl(), nlp_group);
+  copy_vec_to_new_view("sxu", it_curr->get_sxu(), nlp_group);
+  copy_vec_to_new_view("sdl", it_curr->get_sdl(), nlp_group);
+  copy_vec_to_new_view("sdu", it_curr->get_sdu(), nlp_group);
+  copy_vec_to_new_view("yc", it_curr->get_yc(), nlp_group);
+  copy_vec_to_new_view("zl", it_curr->get_zl(), nlp_group);
+  copy_vec_to_new_view("zu", it_curr->get_zu(), nlp_group);
+  copy_vec_to_new_view("vl", it_curr->get_vl(), nlp_group);
+  copy_vec_to_new_view("vu", it_curr->get_vu(), nlp_group);
+
+  //quasi-Newton Hessian approximation 
+  
+  //algorithmic parameters for this state
+  //mu, iteration number
+  const double alg_params[] = {_mu};
```

Comment:
`alg_params` is defined but doesn't appear to be used.

Reply:
will do, I am only half way through here.